tokio/src/io/async_write.rs

```diff
@@ -127,6 +127,51 @@ pub trait AsyncWrite {
     /// This function will panic if not called within the context of a future's
     /// task.
     fn poll_shutdown(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), io::Error>>;
+
+    /// Like [`poll_write`], except that it writes from a slice of buffers.
+    ///
```

Comment:
it might be worth saying something about how this is generally expected to use platform vectored write APIs?

Reply:
I took the documentation for these 2 new methods directly from the standard library. I kind of like that they *don't* mention platform gather IO, and just leave that up to the implementor. :shrug:

Reply:
hmm, that's fair!

Reply:
I think it is worth saying that, if any data is written, the function returns `Ok(n)`. I.e. an error / pending means no data was written.